File: algorithms/2176.count-equal-and-divisible-pairs-in-an-array/solution.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def count_pairs(nums, k)
  count = 0

  for i in 0...nums.size - 1
    for j in i + 1...nums.size
      if nums[i] == nums[j] && i * j % k == 0
        count += 1
      end
    end
  end

  count
end

File: algorithms/2176.count-equal-and-divisible-pairs-in-an-array/solution.rs
impl Solution {
    pub fn count_pairs(nums: Vec<i32>, k: i32) -> i32 {
        let mut count = 0i32;

        for i in 0..nums.len() - 1 {
            for j in i + 1..nums.len() {
                if nums[i] == nums[j] && (i * j) as i32 % k == 0 {
                    count += 1;
                }
            }
        }

        count
    }
}
